### PR TITLE
enhance(Resource/AnomalyDetection):refactor anomaly detection code resource status order cleanup and add tests

### PR DESCRIPTION
**Pull request by Thebarda**
add tests cypress and refactor

### CODE REVIEW

File: centreon/cypress/fixtures/resources/newDetailsAnomalyDetection.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{
```
Comment (bdauria) [inline centreon/cypress/fixtures/resources/newDetailsAnomalyDetection.json:1]: What's the different with the previous one?
Besides that, the correct naming should be: anomalyDetectionDetails and not the other way around

File: centreon/cypress/fixtures/resources/performanceGraphAnomalyDetection.json [hunk 2]
```diff
@@ -0,0 +1,1338 @@
+{
```
Comment (bdauria) [inline centreon/cypress/fixtures/resources/performanceGraphAnomalyDetection.json:1]: should be named: anomalyDetectionPerformanceGraph

File: centreon/cypress/fixtures/resources/resourceListingByTypeAnomalyDetection.json [hunk 3]
```diff
@@ -0,0 +1,106 @@
+{
```
Comment (bdauria) [inline centreon/cypress/fixtures/resources/resourceListingByTypeAnomalyDetection.json:1]: resourceListingWithAnomalyDetectionType

File: centreon/www/front_src/src/Resources/Details/tabs/Graph/index.tsx [hunk 4]
```diff
@@ -63,10 +65,19 @@ const GraphTabContent = ({ details }: TabProps): JSX.Element => {
         <>
           <TimePeriodButtonGroup />
           <ExportablePerformanceGraphWithTimeline
+            interactWithGraph
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Details/tabs/Graph/index.tsx:68]: isInteractive

File: centreon/www/front_src/src/Resources/Details/tabs/Metrics/Metrics.tsx [hunk 5]
```diff
@@ -1,19 +1,18 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Details/tabs/Metrics/Metrics.tsx:1]: is that needed?

File: centreon/www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx [hunk 6]
```diff
@@ -169,12 +164,12 @@ const AnomalyDetectionSlider = ({
     setOpenTooltip(true);
   };
 
-  const setToDefaultValue = (): void => {
+  const handleChangeCheckBox = (event): void => {
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx:167]:
```suggestion
  const handleCheckboxChange = (event): void => {
```

File: centreon/cypress/fixtures/resources/newDetailsAnomalyDetection.json (hunk 1 above)
Comment (Noha-ElAbrouki) [reply centreon/cypress/fixtures/resources/newDetailsAnomalyDetection.json:1]: there is a difference in sensitivity object , the newDetailsAnomalyDetection , this is the result recieved after the user changes the size of envelop  (so different values of sensitivity object) , i check if the value is well updated while the user changed the envelop size 🤔

**Pull request closed without merging**